CMakeLists.txt

```diff
@@ -125,6 +125,13 @@ endif()
 find_package(Threads REQUIRED)
 find_package(ZLIB REQUIRED)
 
+# For some reason Win and Python messes with libcurl
+if(ENABLE_PYTHON_BINDINGS AND WIN32)
+  find_package(CURL REQUIRED)
+  link_libraries(${CURL_LIBRARIES})
+  include_directories(${CURL_INCLUDE_DIRS})
+endif()
+
```

Comment:
NO idea what I'm doing there.. But somehow that's the only way I could make it work when compiling the Python bindings for _and_ on (also one of the very few things I tried..). If that's not set MSVC (and MinGW) errors out with missing links to the `curl` library, similar to https://github.com/valhalla/valhalla/pull/2619#discussion_r498091384 when MinGW didn't find curl..

@mloskot if you have a clue how to that better feel free to comment.. feels very suboptimal.. though I realize you're likely not that interested in the bindings

Reply:
@nilsnolde I don't know why cURL is needed for the bindings.

I have no experience with pybind11, but is this intentional to remove uses of [FindPythonLibs.cmake](https://cmake.org/cmake/help/latest/module/FindPythonLibs.html)?

Reply:
Not asking why it's needed for python (though it shouldn't be needed, maybe some distant dependency from some file..). I'm more asking why MSVC would have problems finding curl as two lines down it again tries to find it. Note, I only need the additional curl lines I added when 1) on Windows & Fedora/MinGW and 2) PYTHON_BINDINGS=ON.

> I have no experience with pybind11, but is this intentional to remove uses of FindPythonLibs.cmake?

Not sure what you mean, but python/pybind has its own CMake where somewhere there's FindPython.cmake by pybind.

Reply:
> I'm more asking why MSVC would have problems finding curl as two lines down it again tries to find it.

Well, it does try to find it, but only `if(ENABLE_HTTP OR ENABLE_DATA_TOOLS)`. 
If that is all false, cURL is not being searched, obviously, but something is still requiring it. 🤷 

All these lines, however, seem redundant indeed
```
# For some reason Win and Python messes with libcurl
if(ENABLE_PYTHON_BINDINGS AND WIN32)
  find_package(CURL REQUIRED)
  link_libraries(${CURL_LIBRARIES})
  include_directories(${CURL_INCLUDE_DIRS})
endif()
```

Reply:
Forgot to mention: I had both variables ON and it still didn't find the curl library when bindings were ON as well.

I changed it slightly to 
```
if(NOT TARGET CURL::CURL)
  add_library(CURL::CURL INTERFACE IMPORTED)
  if(ENABLE_HTTP OR ENABLE_DATA_TOOLS OR ENABLE_PYTHON_BINDINGS)
    if(NOT CURL_FOUND)
      find_package(CURL REQUIRED)
      if (WIN32 AND ENABLE_PYTHON_BINDINGS)
        link_libraries(${CURL_LIBRARIES})
      endif()
    endif()
```

Is that better you think? Guess it's the lazy version instead of fully investigating why that happens and fix the source of the problem. I've seen worse patches though;)

Reply:
it seems better w.r.t. to having single place of `find_package(CURL)` call, but the root of the linking issue is still unknown indeed. I have no idea.